### PR TITLE
Feature/oracle

### PR DESCRIPTION
**Pull request by MarcoDima**
Enable the Service Broker to have more than one ExistingEndpoint.

### CODE REVIEW

File: existing-service/src/main/java/de/evoila/cf/cpi/existing/ExistingServiceFactory.java [hunk 1]
```diff
@@ -76,7 +78,7 @@ public ServiceInstance preCreateInstance(ServiceInstance serviceInstance, Plan p
 
     @Override
     public ServiceInstance postCreateInstance(ServiceInstance serviceInstance, Plan plan) throws PlatformException {
-        serviceInstance.setHosts(existingEndpointBean.getHosts());
+        serviceInstance.setHosts(existingEndpoints.getEndpoints().get(0).getHosts());
```
Comment (jhiemer) [inline existing-service/src/main/java/de/evoila/cf/cpi/existing/ExistingServiceFactory.java]: This is not good practise. I think we agreed on that you have a field in plan which is actually defining which existing Endpoint to use? In Plan there is already a field called `private String endpointName;`. 

So please rewrite this to:
```java
List<ServerAddress> hosts = existingEndpoints.getEndpoints().steam().filter(e -> {
  if (e.getName.equals(plan.getMetadata().getEndpointName()) 
    return true;
  else
   return false;
});
```

**Comment by yremmet** [comment]
Please resolve the merge conflict.

File: existing-service/src/main/java/de/evoila/cf/cpi/existing/ExistingServiceFactory.java (hunk 1 above)
Comment (MarcoDima) [reply existing-service/src/main/java/de/evoila/cf/cpi/existing/ExistingServiceFactory.java]: done. Can you please check?

**Comment by MarcoDima** [comment]
I've merged the develop into the feature/oracle to update all changes. Will still wait before merging it into develop as it's a breaking change

File: existing-service/src/main/java/de/evoila/cf/broker/bean/impl/ExistingEndpoint.java [hunk 2]
```diff
@@ -1,37 +1,37 @@
...
 
+import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 
-/**
- * @author Yannic Remmet, Johannes Hiemer.
- */
-@Service
-@Profile("!pcf")
```
Comment (jhiemer) [inline existing-service/src/main/java/de/evoila/cf/broker/bean/impl/ExistingEndpoint.java]: @MarcoDima I don't think we can remove that

File: existing-service/src/main/java/de/evoila/cf/broker/bean/impl/ExistingEndpoint.java [hunk 3]
```diff
@@ -1,37 +1,37 @@
...
-@ConfigurationProperties(prefix = "existing.endpoint")
-public class ExistingEndpointBeanImpl implements ExistingEndpointBean {
+@Configuration
+@ConfigurationProperties
+public class ExistingEndpoint {
+
+    private String serverName;
```
Comment (jhiemer) [inline existing-service/src/main/java/de/evoila/cf/broker/bean/impl/ExistingEndpoint.java]: For what are we using serverName?

File: existing-service/src/main/java/de/evoila/cf/cpi/existing/ExistingServiceFactory.java [hunk 4]
```diff
@@ -76,7 +81,15 @@ public ServiceInstance preCreateInstance(ServiceInstance serviceInstance, Plan p
...
     public ServiceInstance postCreateInstance(ServiceInstance serviceInstance, Plan plan) throws PlatformException {
-        serviceInstance.setHosts(existingEndpointBean.getHosts());
+		List<ExistingEndpoint> hosts = existingEndpoints.getEndpoints().stream().filter(e -> {
+			if (e.getName().equals(plan.getMetadata().getEndpointName()))
+				return true;
+			else
+				return false;
+			}).collect(Collectors.toList());
+
+    	hosts.stream().forEach(endpoint -> serviceInstance.setHosts(endpoint.getHosts()));
```
[the remark this answers is not in the document]
Comment (jhiemer) [reply existing-service/src/main/java/de/evoila/cf/cpi/existing/ExistingServiceFactory.java]: @MarcoDima ?

**Pull request closed without merging**